### ISSUE TITLE
Throw more useful errors from options functions that cause exceptions

### ISSUE THREAD

**Issue by daniman** [apollostack/react-apollo#303]
```js
const TraceViewPageWithData = graphql(traceViewQuery, {
  options: ({ currentRoute }) => ({
    variables: {
      serviceId: currentRoute.params.serviceId,
    },  }),
})(TraceViewPage);
```

The above options function throws because `currentRoute` is undefined. The error thrown is an opaque apollo error
```
ApolloErrorextraInfo: undefinedgraphQLErrors: undefinedmessage: "Network error: Cannot read property 'replaceChild' of null"networkError: TypeError: Cannot read property 'replaceChild' of null
    at Function.replaceChildWithTree (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:27340:21)
    at Object.dangerouslyReplaceNodeWithMarkup (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:27811:19)
    at Object.dangerouslyReplaceNodeWithMarkup [as replaceNodeWithMarkup] (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:27188:12)
    at ReactCompositeComponentWrapper._replaceNodeWithMarkup (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:32862:31)
    at ReactCompositeComponentWrapper._updateRenderedComponent (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:32852:12)
    at ReactCompositeComponentWrapper._performComponentUpdate (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:32802:10)
    at ReactCompositeComponentWrapper.updateComponent (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:32723:12)
    at ReactCompositeComponentWrapper.receiveComponent (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:32625:10)
    at Object.receiveComponent (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:25247:22)
    at ReactCompositeComponentWrapper._updateRenderedComponent (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:32832:23)message: "Cannot read property 'replaceChild' of null"stack: "TypeError: Cannot read property 'replaceChild' of null↵    at Function.replaceChildWithTree (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:27340:21)↵    at Object.dangerouslyReplaceNodeWithMarkup (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:27811:19)↵    at Object.dangerouslyReplaceNodeWithMarkup [as replaceNodeWithMarkup] (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:27188:12)↵    at ReactCompositeComponentWrapper._replaceNodeWithMarkup (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:32862:31)↵    at ReactCompositeComponentWrapper._updateRenderedComponent (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:32852:12)↵    at ReactCompositeComponentWrapper._performComponentUpdate (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:32802:10)↵    at ReactCompositeComponentWrapper.updateComponent (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:32723:12)↵    at ReactCompositeComponentWrapper.receiveComponent (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:32625:10)↵    at Object.receiveComponent (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:25247:22)↵    at ReactCompositeComponentWrapper._updateRenderedComponent (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:32832:23)"__proto__: Errorstack: "Error↵    at new ApolloError (http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:56363:22)↵    at http://localhost:3000/packages/modules.js?hash=1858f0a59ec058881704e047599f959cb1ba6e8c:56234:28↵    at http://localhost:3000/packages/meteor.js?hash=ae8b8affa9680bf9720bd8f7fa112f13a62f71c3:1105:22"__proto__: Error(anonymous function) @ modules.js?hash=1858f0a…:56234(anonymous function) @ meteor.js?hash=ae8b8af…:1105```

**Comment by tmeasday** [comment]
See https://github.com/apollostack/react-apollo/issues/347, which I think is probably a simpler version of this same error

**Comment by tmeasday** [comment]
I think this should be fixed by `apollo-client@0.5.9`. Let me know if not @daniman
